pkg/onboarding/usecases/shortcode.go

```diff
@@ -0,0 +1,155 @@
+package usecases
```

Comment:
I strongly think this logic belongs to `engagement` service. Feel free to convince me otherwise.

Your last argument was that we'll use short code for onboarding, but after reading the code I have not seen a single point where a new user is being onboarded.